api/vcs/v1/vcs.proto

```diff
@@ -48,11 +48,16 @@ message GetFileResponse {
 message GetCommitRequest {
   // the full path to the repository
   string repositoryURL = 1;
-  // the vcs ref to get the file from
-  string ref = 2;
+  // the vcs refs to get the file from
+  repeated string refs = 2;
```

Comment:
I would likely extend the API by introducing a new method: `ListCommits`

[The standard approach](https://protobuf.dev/programming-guides/proto3/#updating) is to extend the API and preserve the former behavior for the old schema.

Please consider implementing the changes in a backward-compatible way; otherwise, we can't avoid service interruptions during the rollout, as the client (frontend) is updated independently of the backend